web/src/engine/main/src/keymanEngine.ts

```diff
@@ -137,31 +137,57 @@ export default class KeymanEngine<
     });
 
     this.contextManager.on('keyboardchange', (kbd) => {
-      this.refreshModel();
-      this.core.activeKeyboard = kbd?.keyboard;
-
-      this.legacyAPIEvents.callEvent('keyboardchange', {
-        internalName: kbd?.metadata.id ?? '',
-        languageCode: kbd?.metadata.langId ?? ''
-      });
-
       // Hide OSK and do not update keyboard list if using internal keyboard (desktops).
       // Condition will not be met for touch form-factors; they force selection of a
       // default keyboard.
       if(!kbd) {
         this.osk.startHide(false);
       }
 
-      if(this.osk) {
-        this.osk.setNeedsLayout();
-        this.osk.activeKeyboard = kbd;
-        this.osk.present();
+      const earlyBatchClosure = () => {
```

Comment:
So ... on variable names ... earlyBatchClosure doesn't tick the right boxes for me.

My feeling on the use of the word 'closure' is that we should not be using it. It's like saying 'oskVariable' instead of 'osk'. It should be obvious from the variable name that it is a closure.

I am not sure what the 'batch' is here, and why this is 'early'? Can we find a name that describes what the function does rather than when it gets called?

Reply:
Would something like `prepareSwap()` and `finalizeSwap()` be better?

Reply:
I'm not sure what the closures are actually doing -- are they swapping something?